app/lib/activitypub/activity/create.rb

```diff
@@ -86,15 +86,18 @@ def process_mention(tag, status)
   end
 
   def process_emoji(tag, _status)
-    return if tag['name'].blank? || tag['href'].blank?
+    return if tag['name'].blank? || tag['icon'].blank? || tag['icon']['url'].blank?
 
     shortcode = tag['name'].delete(':')
+    image_url = tag['icon']['url']
+    uri       = tag['id']
+    updated   = tag['updated']
     emoji     = CustomEmoji.find_by(shortcode: shortcode, domain: @account.domain)
 
-    return if !emoji.nil? || skip_download?
+    return if skip_download? || (!emoji.nil? && emoji.updated_at >= updated && emoji.image_remote_url == image_url)
```

Comment:
this would probably be clearer as two return statements

Reply:
do we need the `emoji.image_remote_url == image_url` check? wouldn't the save just be a no-op?